### ISSUE TITLE
Editing and saving input values in dashboard causes error

### ISSUE THREAD

**Issue by reyery** [architecture-building-systems/CityEnergyAnalyst#2033]
There seems to be a problem when editing input values in the dashboard and saving them.
Changing a value (e.g. edit `height_ag` and click save) will cause an error.

This happens because the program is trying to convert the coordinates of the polygons but is unable to find the original coordinate system of the file.
```
Traceback (most recent call last):
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask\app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask\app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask\app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\flask\app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\CityEnergyAnalyst\cea\interfaces\dashboard\inputs\routes.py", line 105, in route_geojson
    table_df = table_df.to_crs(epsg=4326)  # make sure that the geojson is coded in latitude / longitude
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\geopandas\geodataframe.py", line 459, in to_crs
    geom = df.geometry.to_crs(crs=crs, epsg=epsg)
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\geopandas\geoseries.py", line 304, in to_crs
    proj_in = pyproj.Proj(self.crs, preserve_units=True)
  File "C:\Users\Admin\Documents\CityEnergyAnalyst\Dependencies\Python\lib\site-packages\pyproj\__init__.py", line 358, in __new__
    return _proj.Proj.__new__(self, projstring)
  File "_proj.pyx", line 84, in _proj.Proj.__cinit__ (_proj.c:1170)
RuntimeError: no arguments in initialization list
```

For some reason, when saving the new dataframe to file, it causes the `.prj` file (the place where coordinate system information is stored) to be overwritten by an empty file.

https://github.com/architecture-building-systems/CityEnergyAnalyst/blob/626f89dffd5798a646148b0fa260b253d8ffa2d2/cea/interfaces/dashboard/inputs/routes.py#L144-L145

Searching for a solution [online](https://stackoverflow.com/questions/54963638/geopandas-to-file-gives-blank-prj-file) seems to suggest that. it is due to the installation of `geopandas` and it's dependencies (i.e. `fiona` and `gdal`)

**Comment by reyery** [comment]
@daren-thomas any clues for why this is happening? I want to test the installation theory but am not sure how to do it with the new cea installation since installing `fiona` and `gdal` using `pip` is quite complicated.

**Comment by daren-thomas** [comment]
I noticed that installations running with the CEA Console have a lot of fiona messages about GDAL paths etc.

```
DEBUG:fiona:Creating a not-responsible GDALEnv in drivers()
WARNING:Fiona:GDAL data files not located, GDAL_DATA not set
WARNING:Fiona:PROJ data files not located, PROJ_LIB not set
WARNING:Fiona:GDAL data files not located, GDAL_DATA not set
WARNING:Fiona:PROJ data files not located, PROJ_LIB not set
DEBUG:Fiona:Projection not found (cogr_crs was NULL)
DEBUG:Fiona:Projection not found (cogr_crs was NULL)
```

This page has some hints: https://stackoverflow.com/questions/14444310/how-to-set-the-gdal-data-environment-variable-to-point-to-the-directory-containi

I did a quick test with the following: Open `$INSTDIR\dashboard.bat` and edit it to be like this:

```
SET PATH=C:\Users\darthoma\Documents\CityEnergyAnalyst\Dependencies\Python;C:\Users\darthoma\Documents\CityEnergyAnalyst\Dependencies\Python\Scripts;C:\Users\darthoma\Documents\CityEnergyAnalyst\Dependencies\Daysim;%PATH%
SET PYTHONHOME=C:\Users\darthoma\Documents\CityEnergyAnalyst\Dependencies\Python
SET RAYPATH=C:\Users\darthoma\Documents\CityEnergyAnalyst\Dependencies\Daysim
SET GDAL_DATA=C:\Users\darthoma\Documents\CityEnergyAnalyst\Dependencies\Python\Library\share\gdal
SET PROJ_LIB=c:\Users\darthoma\Documents\CityEnergyAnalyst\Dependencies\Python\Library\share
C:\Users\darthoma\Documents\CityEnergyAnalyst\Dependencies\Python\python.exe -m cea.interfaces.cli.cli dashboard
```

(update the paths to reflect your user) - note the setting of `GDAL_DATA`  and `PROJ_LIB`. After that, I was able to save changes to the Zone input file. There are still errors, but they relate to the network files not being there or something. I think that might be an unrelated issue.

Does doing this work for you? If so, I'll update the installer to reflect these changes for release 2.15. Also, we can add this issue as a known issue with a fix!

**Comment by reyery** [comment]
@daren-thomas can confirm that setting `GDAL_DATA` and `PROJ_LIB` solves the problem when running from the dashboard shortcut. Is it possible to add these environment variables from the installer so that it fixes the problem for users that start it from cmd line or pycharm?

**Comment by daren-thomas** [comment]
This should be fixed with Release 2.15. Would you mind trying out the installer please? https://github.com/architecture-building-systems/CityEnergyAnalyst/releases/tag/v2.15

**Comment by reyery** [comment]
@daren-thomas I can confirm that this issue is fixed with Release 2.15, when running from dashboard shortcut or cmd line. Closing issue.
